TowerForge/TowerForge/GameModule/GameModes/GameMode.swift

```diff
@@ -7,7 +7,18 @@
 
 import Foundation
 
-struct GameResult {
+protocol GameResult {
+    var variable: String { get  }
+    var value: String { get }
+}
+
+struct LeaderboardResult: GameResult {
+    var variable: RankType.RawValue
+    var result: Double
+    var value: String
```

Comment:
Also, it might be better to store the information in the double and make value a computed property from result.

Reply:
but i was thinking cos it conforms to `GameResult` so it has to have the `value` but I will think about how to refactor this.